sycl/source/detail/builtins_relational.cpp

```diff
@@ -458,9 +458,15 @@ __SYCL_EXPORT rel_res_t sycl_host_SignBitSet(s::cl_float x) __NOEXC {
 __SYCL_EXPORT rel_res_t sycl_host_SignBitSet(s::cl_double x) __NOEXC {
   return std::signbit(x);
 }
+#ifndef __GNUC__
+// GCC 11.3.0 (and friends) has an internal compiler error when this function is
+// passed to the macro declaration below. ( MAKE_1V ... 4 ... ) Fortunately, we
+// don't actually need it when compiling the SYCL library itself.
+// TODO: switch to templates instead of these overload expansions.
 __SYCL_EXPORT s::cl_int __vSignBitSet(s::cl_float x) __NOEXC {
   return -static_cast<s::cl_int>(std::signbit(x));
 }
+#endif
```

Comment:
I tested it locally and I believe we can do this without breaking ABI, by simply doing the redirection that we would do anyway.

```suggestion
__SYCL_EXPORT s::cl_int __vSignBitSet(s::cl_float x) __NOEXC {
  return static_cast<s::cl_int>(__vSignBitSet(static_cast<s::cl_double>(x)));
}
```

Alternatively, if we want to preserve the old version when building with a non-GCC compiler, we can do something like:

```suggestion
__SYCL_EXPORT s::cl_int __vSignBitSet(s::cl_float x) __NOEXC {
#ifdef __GNUC__
  return static_cast<s::cl_int>(__vSignBitSet(static_cast<s::cl_double>(x)));
#else
  return -static_cast<s::cl_int>(std::signbit(x));
#endif
}
```

Feel free to leave a comment about the work-around and the TODO to move them.